lib/panda_doc/document.rb

```diff
@@ -8,6 +8,10 @@ def create(data)
       respond(ApiClient.request(:post, "/documents", **data))
     end
 
+    def update(uuid, **data)
+      respond(ApiClient.request(:patch, "/documents/#{uuid}", **data))
```

Comment:
```suggestion
      respond(ApiClient.request(:patch, "/documents/#{uuid}", **data), type: :base)
```

I think that should build an empty response instead of `:document`.

Reply:
Hmm, having a bit of trouble figuring out how to resolve this. When I use this client against the real PandaDoc API, the empty response body (or any non-JSON response) doesn't get parsed to an empty hash/map, and tries to convert the raw response body

```ruby
PandaDoc::Document.update("...", metadata: { test: "hello" } )
=> returns 204: ""

raises: [PandaDoc::Objects::Base.new] can't convert String into Hash (Dry::Struct::Error)
```

Reply:
Ah, I see. In that case it would make sense to introduce a dedicated object for that kind of response in https://github.com/opti/panda_doc/tree/main/lib/panda_doc/objects. Since it's an empty response anyway, a `null_object` could be a good candidate for that:

```ruby
# frozen_string_literal: true

module PandaDoc
  module Objects
    class NullObject
      def initialize(_response_body)
      end
    end
  end
end
```

```ruby
# lib/panda_doc/document.rb
def update(uuid, **data)
  respond(
    ApiClient.request(:patch, "/documents/#{uuid}", **data),
    type: :null_object
  )
end
```